qiskit/aqua/algorithms/factorizers/shor.py

```diff
@@ -84,7 +85,7 @@ def __init__(self,
 
         self._a = a
 
-        self._ret = {'factors': []}
+        self._ret = AlgorithmResult({"factors": [], "results": {}})
```

Comment:
I think this is fine for now. Just as an FYI the intent of the algorithm grouping is to have a defined common result, so such algorithms can be used interchangeably, such as in minimum eigensolvers, where the result has defined properties. Properties can be documented and show up in autocomplete in IDEs etc. At present we maintained dictionary keys for backwards compatibility. This changeover is work in progress that we started for the last release.

Reply:
In that case, do you think it would be out of the scope of this PR to create a `FactorizerResult`/`ShorResult` for this algorithm?

Reply:
I think this is ok for now as you have it.